### PR TITLE
fix: [CDS-78731]: Adding the Schema Validation Check

### PR DESCRIPTION
**Pull request by vinit1968**
## Describe your changes
There was missing schema validation check for validateMultiServices .Due to which pipeline was able to save without any services selected.So Schema validation check was added so now we can’t save the pipeline without services as empty.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53452)

### CODE REVIEW

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger messagecheck

**Comment by vinit1968** [comment]
trigger messagecheck

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger checkstylecodeformat

**Comment by vinit1968** [comment]
trigger messagecheck

**Comment by vinit1968** [comment]
trigger messagecheck

**Comment by vinit1968** [comment]
trigger messagecheck

**Comment by vinit1968** [comment]
trigger checkstylecodeformat

**Comment by vinit1968** [comment]
trigger uts

**Review by naman-goenka — APPROVED** [review]

**Comment by vinit1968** [comment]
trigger ut7

**Comment by vinit1968** [comment]
UnitTests-4: trigger ut4

**Comment by vinit1968** [comment]
trigger ut4

**Comment by tchaurasiya** [comment]
Lgtm, Please add the pr description

**Review by tchaurasiya — DISMISSED** [review]

**Review by lovish1999 — APPROVED** [review]
LGTM

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/filters/DeploymentStageFilterJsonCreatorV2.java [hunk 1]
```diff
@@ -138,6 +138,11 @@ private void validate(FilterCreationContext filterCreationContext, DeploymentSta
 
   private void validateMultiServices(
       FilterCreationContext filterCreationContext, DeploymentStageConfig deploymentStageConfig) {
+    if (ParameterField.isNotNull(deploymentStageConfig.getServices().getValues())
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/filters/DeploymentStageFilterJsonCreatorV2.java:141]: Should we add null check for `deploymentStageConfig.getServices()` also ?

Comment (lovish1999) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/filters/DeploymentStageFilterJsonCreatorV2.java:141]: Ok, we do not need it as this check is there before the method is called.

**Review by lovish1999 — APPROVED** [review]

**Review by rafaelmart1ns — APPROVED** [review]

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger ut6

**Comment by vinit1968** [comment]
trigger ut9

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger uts

**Comment by vinit1968** [comment]
trigger ut1

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger ut7

**Comment by vinit1968** [comment]
trigger ut1

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger ut1

**Comment by vinit1968** [comment]
trigger ut4

**Comment by vinit1968** [comment]
trigger ut4

**Comment by naman-goenka** [comment]
trigger ut1

**Comment by vinit1968** [comment]
trigger ut1

**Review by naman-goenka — APPROVED** [review]

**Review by lovish1999 — APPROVED** [review]

**Comment by vinit1968** [comment]
trigger review

**Review by yogesh-chauhan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53452)* status: 4 file(s) to review.

**Pull request merged**